app/org/sagebionetworks/bridge/validators/ConsentSignatureValidator.java

```diff
@@ -22,11 +30,28 @@ public boolean supports(Class<?> clazz) {
     @Override
     public void validate(Object target, Errors errors) {
         ConsentSignature sig = (ConsentSignature) target;
+        
         if (Strings.isNullOrEmpty(sig.getName())) {
             errors.rejectValue("name", CANNOT_BE_BLANK);
         }
-        if (Strings.isNullOrEmpty(sig.getBirthdate())) {
-            errors.rejectValue("birthdate", CANNOT_BE_BLANK);
+        if (minAgeOfConsent > 0) {
+            if (Strings.isNullOrEmpty(sig.getBirthdate())) {
+                errors.rejectValue("birthdate", CANNOT_BE_BLANK);
+            } else {
+                LocalDate birthdate = parseBirthday(errors, sig.getBirthdate());
+                if (birthdate != null) {
+                    LocalDate now = LocalDate.now();
+                    Period period = new Period(birthdate, now);
+
+                    if (period.getYears() < minAgeOfConsent) {
+                        String message = String.format(TOO_YOUNG, minAgeOfConsent);
+                        errors.rejectValue("birthdate", message);
+                    }
+                }
+            }
+        } else if (sig.getBirthdate() != null) {
+            // Just verify it parses because it wasn't required, but they supplied it anyway
+            parseBirthday(errors, sig.getBirthdate());
```

Comment:
Instead of parsing birthdate in 2 different branches, considering parsing the birthdate once, before the minAge check, and saving the result so you don't need to parse again.

Reply:
Rearranged